pkg/kind/dashboard/kind_gen.go

```diff
@@ -0,0 +1,91 @@
+// This will be autogenerated, same as pkg/coremodel/dashboard/dashboard_gen.go is today
+
+package dashboard
+
+import (
+	"embed"
+
+	"github.com/grafana/grafana/pkg/framework/kind"
+	"github.com/grafana/thema"
+	"github.com/grafana/thema/vmux"
+)
+
+//go:embed kind.cue
+var cueFS embed.FS
+
+// TODO standard generated docs
+type Kind struct {
+	lin    thema.ConvergentLineage[*Entity]
+	meta   kind.CoreStructuredMeta
+	jendec vmux.Endec
+	valmux vmux.ValueMux[*Entity]
+}
```

Comment:
It is expected that kind authors can attach additional handwritten methods by just adding another file in this directory. The generated registry that exposes them (successor to [this one](https://github.com/grafana/grafana/blob/ae927eab73832b37b9a7023b079a538fa0cf5a60/pkg/framework/coremodel/registry/registry_gen.go)) will expose each kind's generated `Kind` struct directly with a single method (same as [this](https://github.com/grafana/grafana/blob/ae927eab73832b37b9a7023b079a538fa0cf5a60/pkg/framework/coremodel/registry/registry_gen.go#L44)), while also exposing all of them via interfaces (same as [this](https://github.com/grafana/grafana/blob/main/pkg/framework/coremodel/registry/provide.go#L46)), so getting at the kind-specific methods will be easy for anything working directly with a particular kind.

That said, `pkg/kind` will be import-restricted via ruleguard to disallow importing most things inside `pkg` (at minimum, nothing under `pkg/service`), in the interest of keeping this entire subsystem light and externally importable. The effect should be that stateless, storage- and api-agnostic logic can be declared here, and other stuff will need to be layered on in other packages, analogous to the layering we've already done [with the playlist service](https://github.com/grafana/grafana/blob/main/pkg/services/playlist/model.go#L25-L27).